build.gradle

```diff
@@ -32,9 +34,124 @@ dependencies {
+// EXPERIMENTAL: Builds a JAR-based plugin file with dependencies embedded.
+task pluginJar(type: Jar) {
+    destinationDir = file("pkg")
+
+    configurations {
+        pluginJarArtifact
+    }
+
+    artifacts {
+        pluginJarArtifact file: pluginJar.archivePath, builtBy: pluginJar
+    }
+
+    manifest {
+        attributes 'Embulk-Plugin-Spi-Version': "0",
+                   'Embulk-Plugin-Main-Class': "org.embulk.filter.copy.CopyFilterPlugin",
+                   'Implementation-Title': project.name,
+                   'Implementation-Version': version
+    }
+
+    from {
+        // "provided" dependencies are excluded as they are provided at runtime by the Embulk core.
+        def embedded = configurations.runtime - configurations.provided
+
+        // Dependencies are picked up with extracting ".jar" files.
+        embedded.collect { ( it.isFile() && it.name.endsWith(".jar") ) ? zipTree(it) : it }
+    }
+
+    // Signature files of dependencies are excluded as they cause SecurityException.
+    exclude("META-INF/*.DSA")
+    exclude("META-INF/*.RSA")
+    exclude("META-INF/*.SF")
+
+    with jar
+}
+clean { delete "pkg" }
+
+def cleanupPomInPluginJar = {
+    // All dependencies except for "org.embulk:embulk-core" are removed.
+    whenConfigured { pom ->
+        pom.dependencies = pom.dependencies.findAll { dependency ->
+            ( dependency.groupId == "org.embulk" && dependency.artifactId == "embulk-core" )
+        }
+    }
+
+    // The dependency "org.embulk:embulk-core" is "provided" for plugin JARs.
+    whenConfigured { pom ->
+        pom.dependencies.find { dependency ->
+            ( dependency.groupId == "org.embulk" && dependency.artifactId == "embulk-core" )
+        }.scope = "runtime"
+    }
+}
+
+// EXPERIMENTAL: Uploads the JAR-based plugin build in "pluginJar".
+task uploadPluginJar(type: Upload) {
+    configuration = configurations.pluginJarArtifact
+
+    repositories {
+        mavenDeployer {
+            configurations {
+                pluginJarDeployer
+            }
+
+            dependencies {
+                pluginJarDeployer "net.trajano.wagon:wagon-git:2.0.4"
+            }
+
+            configuration = configurations.pluginJarDeployer
+
+            // "gradle assemble" fails at "getProperty" if the properties are not defined.
+            if (hasProperty("io.github.embulk.plugins.embulk.org.username") &&
+                hasProperty("io.github.embulk.plugins.embulk.org.access_token")) {
+                repository(url: "git:https://github.com/embulk/plugins.embulk.org.git?gh-pages#") {
+                    // Set your GitHub username and personal access token in your ~/.gradle/gradle.properties.
+                    // Do not set your GitHub main password!
+                    authentication(userName: getProperty("io.github.embulk.plugins.embulk.org.username"),
+                                   password: getProperty("io.github.embulk.plugins.embulk.org.access_token"))
```

Comment:
Is it enough privileges?

![screen shot 2017-07-01 at 22 06 26](https://user-images.githubusercontent.com/4525500/27762419-ae937838-5eac-11e7-9eea-abafbdea5eec.png)